tests/foreman/api/test_repository.py

```diff
@@ -1500,6 +1500,38 @@ def test_missing_content_id(self, repo, function_sca_manifest_org, target_sat):
         )
         assert 'Candlepin job status: SUCCESS' in output, 'Failed to refresh manifest'
 
+    @pytest.mark.skipif(
+        (not settings.robottelo.REPOS_HOSTING_URL), reason='Missing repos_hosting_url'
+    )
+    def test_positive_sync_rpm_missing_filelists(self, target_sat):
+        """Synchronize repository with RPMs that have missing filelists with SSL verification disabled.
+
+        :id: 6a5a3d4d-f9b1-4a2c-8e5d-2b9c4d8f7e6a
+
+        :expectedresults: Repository sync should complete successfully even when
+            RPM packages are missing filelists metadata and SSL verification is disabled.
+
+        :CaseImportance: Medium
+
+        :verifies: SAT-30999
+        """
+        org = target_sat.api.Organization().create()
+        product = target_sat.api.Product(organization=org).create()
+        repo = target_sat.api.Repository(
+            product=product, 
+            content_type='yum',
+            url=settings.repos.rpm_missing_filelists.url,
+            verify_ssl_on_sync=False
```

Comment:
It is not a good practice to hardcode whether SSL/TLS is enabled or disabled. There should be a config for that. 

Looking at other places where we sync Products that are mapped to our custom repos, I don't see us using the `verify_ssl_on_sync` option. Would you happen to know if we need to use it in this case explicitly?

Reply:
IMHO it was necessary for repo sourced on the local FS. Shouldn't be needed for `url=settings.repos.rpm_missing_filelists.url`.
Can you try rerun w/o that @vijaysawant ?